### PR TITLE
Use DOMContentLoaded and fix sharebymail loading issue

### PR DESCRIPTION
**Pull request by skjnldsv**
After https://github.com/nextcloud/server/pull/20949

### CODE REVIEW

File: apps/files_sharing/src/services/ConfigService.js [hunk 1]
```diff
@@ -178,8 +178,11 @@ export default class Config {
 	 * @memberof Config
 	 */
 	get isMailShareAllowed() {
-		return OC.getCapabilities()['files_sharing']['sharebymail'] !== undefined
-			&& OC.getCapabilities()['files_sharing']['public']['enabled'] === true
+		const capabilities = OC.getCapabilities()
+		// eslint-disable-next-line camelcase
+		return capabilities?.files_sharing?.sharebymail !== undefined
+			// eslint-disable-next-line camelcase
+			&& capabilities?.files_sharing?.public?.enabled === true
```
Comment (skjnldsv) [inline apps/files_sharing/src/services/ConfigService.js:185]: For you @rullzer

**Comment by skjnldsv** [comment]
Failure unrelated :)
`app-comments.feature`

**Review by juliusknorr — APPROVED** [review]
Seems to do the trick with the random loading issues.

**Review by georgehrke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Oh! This makes the integration tests more reliable! 🚀

**Pull request merged**